src/Avalonia.ReactiveUI/TransitioningUserControl.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) The Avalonia Project. All rights reserved.
+// Licensed under the MIT license. See licence.md file in the project root for full license information.
+
+using System;
+using Avalonia.Animation;
+using Avalonia.Controls;
+using Avalonia.Styling;
+
+namespace Avalonia.ReactiveUI
+{
+    /// <summary>
+    /// A ContentControl that animates the transition when its content is changed.
+    /// </summary>
+    public class TransitioningUserControl : UserControl
+    {
+        /// <summary>
+        /// <see cref="AvaloniaProperty"/> for the <see cref="FadeInAnimation"/> property.
+        /// </summary>
+        public static readonly AvaloniaProperty<IAnimation> FadeInAnimationProperty =
+            AvaloniaProperty.Register<TransitioningUserControl, IAnimation>(nameof(DefaultContent),
+                CreateOpacityAnimation(0d, 1d, TimeSpan.FromSeconds(0.25)));
+
+        /// <summary>
+        /// <see cref="AvaloniaProperty"/> for the <see cref="FadeOutAnimation"/> property.
+        /// </summary>
+        public static readonly AvaloniaProperty<IAnimation> FadeOutAnimationProperty =
+            AvaloniaProperty.Register<TransitioningUserControl, IAnimation>(nameof(DefaultContent),
+                CreateOpacityAnimation(1d, 0d, TimeSpan.FromSeconds(0.25)));
+
+        /// <summary>
+        /// <see cref="AvaloniaProperty"/> for the <see cref="DefaultContent"/> property.
+        /// </summary>
+        public static readonly AvaloniaProperty<object> DefaultContentProperty =
+            AvaloniaProperty.Register<TransitioningUserControl, object>(nameof(DefaultContent));
+        
+        /// <summary>
+        /// Gets or sets the animation played when content appears.
+        /// </summary>
+        public IAnimation FadeInAnimation
```

Comment:
We actually have an interface to represent page transitions: `IPageTransition`. Would that be suitable here instead of having separate fade in/out animations?

Or does this API come from RxUI? I couldn't find `TransitioningUserControl` there, only `TransitioningContentControl`. Why is it a `UserControl` in Avalonia and a `ContentControl` in WPF?

Reply:
Interesting, looks like it'd be rather easy to rewrite the existing implementation of the transitioning control to use `IPageTransition` with `CrossFade` set by default, added this to my todo list. Considering `TransitioningUserControl`, I experienced several issues with default templates when inheriting from a regular `ContentControl`, but don't remember exactly. Will try to use `ContentControl` again as well.

Reply:
Just changed the `TransitioningContentControl` to derive from `ContentControl`, so far works good! But when changing the `TransitioningContentControl` to use the `IPageTransition` interface and the  `CrossFade` implementation, the transition loses its smoothness, not sure what causes this. The view disappears smoothly, but appears without any animation. Calling the `IPageTransition` interface as such:
```cs
// Animation works good when the content disappears.
await PageTransition.Start((Visual)base.Content, null, true);
base.Content = content;

// Animation doesn't work, opacity changes from 0 to 1 after the specified delay.
await PageTransition.Start(null, (Visual)base.Content, true);
```

Reply:
Resolved this by changing these two lines from `0d` to `1d`, probably a bug with `CrossFade` https://github.com/AvaloniaUI/Avalonia/blob/a8d5814a10198b0c42fff3a5b13d24eccb77703a/src/Avalonia.Visuals/Animation/CrossFade.cs#L64 https://github.com/AvaloniaUI/Avalonia/blob/a8d5814a10198b0c42fff3a5b13d24eccb77703a/src/Avalonia.Visuals/Animation/CrossFade.cs#L67

Reply:
@worldbeater Yeah, that's definitely an error, thanks for pointing it out :D